backend/tests/legacy/test_api_deposits.py

```diff
@@ -0,0 +1,158 @@
+def test_deposit_basics(
+    client: Client, deployer: UserAccount, ua_alice: UserAccount, ua_bob: UserAccount
+):
+    # Check effective deposit before first GLM lock
+    (
+        total_effective_estimated_before_lock,
+        response_code,
+    ) = client.get_total_effective_estimated()
+    assert int(total_effective_estimated_before_lock["totalEffective"]) == 0
+    assert response_code == 200
+
+    user_deposit1_before_lock, response_code = client.get_user_deposit(
+        ua_alice.address, 1
+    )
+    assert int(user_deposit1_before_lock["effectiveDeposit"]) == 0
+    assert response_code == 200
+
+    (
+        user_estimated_deposit_before_lock,
+        response_code,
+    ) = client.get_user_estimated_effective_deposit(ua_alice.address)
+    assert int(user_estimated_deposit_before_lock["effectiveDeposit"]) == 0
+    assert response_code == 200
+
+    # Lock GML for one account
+    alice_GLM_budget = 10000
+    alice_first_lock = 8000
+    deployer.transfer(ua_alice, alice_GLM_budget)
+    ua_alice.lock(alice_first_lock)
+
+    indexed_height = client.wait_for_height_sync()
+    print(f"indexed blockchain height: {indexed_height}")
+
+    # Check effective deposit after first GLM lock
+    user_deposit_after_lock1, _ = client.get_user_deposit(ua_alice.address, 1)
+    assert (
+        0
+        < int(user_deposit_after_lock1["effectiveDeposit"])
+        < w3.to_wei(alice_first_lock, "ether")
+    )
+
+    user_estimated_deposit_after_lock1, _ = client.get_user_estimated_effective_deposit(
+        ua_alice.address
+    )
+    assert (
+        0
+        < int(user_estimated_deposit_after_lock1["effectiveDeposit"])
+        < w3.to_wei(alice_first_lock, "ether")
+    )
+
+    total_effective_estimated_after_lock1, _ = client.get_total_effective_estimated()
+    assert (
+        0
+        < int(total_effective_estimated_after_lock1["totalEffective"])
+        < w3.to_wei(alice_first_lock, "ether")
+    )
+
+    # Second GLM lock
+    ua_alice.lock(alice_GLM_budget - alice_first_lock)
+
+    indexed_height = client.wait_for_height_sync()
+    print(f"indexed blockchain height: {indexed_height}")
+
+    # Check effective deposit after second GLM lock
+    user_deposit_after_lock2, _ = client.get_user_deposit(ua_alice.address, 1)
+    assert int(user_deposit_after_lock2["effectiveDeposit"]) > int(
+        user_deposit_after_lock1["effectiveDeposit"]
+    )
+
+    user_estimated_deposit_after_lock2, _ = client.get_user_estimated_effective_deposit(
+        ua_alice.address
+    )
+    assert int(user_estimated_deposit_after_lock2["effectiveDeposit"]) > int(
+        user_estimated_deposit_after_lock1["effectiveDeposit"]
+    )
+
+    total_effective_estimated_after_lock2, _ = client.get_total_effective_estimated()
+    assert int(total_effective_estimated_after_lock2["totalEffective"]) > int(
+        total_effective_estimated_after_lock1["totalEffective"]
+    )
+
+    # forward time to the beginning of the epoch 2
```

Comment:
these tests won't be very valuable if we test only for epoch 1 - I think we should implement them from the start in a way that the epoch number is configurable and set to the current epoch we have in Octant - 4 in this case

Reply:
I think it is a good idea but let's mark it as future improvement.

Reply:
In this case please add TODO ticket with ticket ticket number in here. Otherwise it will most certainly get lost.
```
# TODO OCT-9999 {explanation}
```

Reply:
is it a hard to do it now? Later we'll have a lot of places to change and this change might consume a lot of time, now we can extract the functions to move time with a proper starting point and then reuse it accross the tests

Reply:
and also what @aziolek pointed out - it's likely that we'll forget about it

Reply:
OCT-1640 is about using the current epoch, which is not the same.

API should work for all the epochs, including previous ones. Otherwise, this can lead to situations where API works for epoch 10, but does not work for epoch 1, breaking local and E2E environments.

PRs are no interconflicted and are not in a rush of being merged. I would push for this improvement here, currently it's like splitting hairs.

If you wish to extract this into separate ticket them please leave a TODO in code.